recipes/serializers.py

```diff
@@ -0,0 +1,50 @@
+from django.core.exceptions import ObjectDoesNotExist
+from .models import Recipe, Ingredient, User
+from rest_framework import serializers
+from pytz import timezone
+
+
+class RecipeSerializer(serializers.ModelSerializer):
+    ingredients = serializers.ListField(child=serializers.CharField())
+    difficulty = serializers.IntegerField(min_value=1, max_value=4)
+
+    class Meta:
+        model = Recipe
+        fields = ('title', 'content', 'vote_points', 'vote_count', 'like_count', 'difficulty', 'ingredients')
+
+    def create(self, data):
+        request = self.context.get("request")
+        author = request.user
+        recipe = Recipe(title=data["title"], content=data["content"],
+                        author=author)
+        recipe.save()
+        for ingredient_name in data["ingredients"]:
+            lookup_name = ingredient_name.lower()
+            lookup_name = lookup_name.replace(" ", "_")
+            try:
+                ingredient = Ingredient.objects.get(lookup_name=lookup_name)
+                recipe.ingredients.add(ingredient)
+            except ObjectDoesNotExist:
+                ingredient = Ingredient(name=ingredient_name, lookup_name=lookup_name)
+                ingredient.save()
+                recipe.ingredients.add(ingredient)
+
+        recipe.difficulty = str(data["difficulty"])
+        recipe.save()
```

Comment:
why do you save recipe twice?

Reply:
Since Recipe have a many to many relationship with Ingredient, it has to have an id before adding ingredient instances to it. That's why I have used .save() twice.

Reply:
This can be achieved without saving recipe twice.